### ISSUE TITLE
login via "GitHub" or "Google" in the demo seems to be broken.

### ISSUE THREAD

**Issue by cbleek** [cross-solution/YAWIK#257]
Error Message: No adapter registered for key google in the adapter map.

```
2016-08-02 16:23:18 EXCEPTION No adapter registered for key google in the adapter map. in /home/cbleek/Projects/YAWIK/module/Auth/src/Auth/Controller/Plugin/SocialProfiles.php on line 80
[Trace]
        File: /home/cbleek/Projects/YAWIK/module/Auth/src/Auth/Controller/Plugin/SocialProfiles.php
        Line: 43
    Function: getAdapter
       Class: Auth\Controller\Plugin\SocialProfiles
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/module/Auth/src/Auth/Adapter/HybridAuth.php
        Line: 110
    Function: fetch
       Class: Auth\Controller\Plugin\SocialProfiles
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-authentication/src/AuthenticationService.php
        Line: 110
    Function: authenticate
       Class: Auth\Adapter\HybridAuth
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/module/Auth/src/Auth/AuthenticationService.php
        Line: 82
    Function: authenticate
       Class: Zend\Authentication\AuthenticationService
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/module/Auth/src/Auth/Controller/IndexController.php
        Line: 201
    Function: authenticate
       Class: Auth\AuthenticationService
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-mvc/src/Controller/AbstractActionController.php
        Line: 82
    Function: loginAction
       Class: Auth\Controller\IndexController
        Type: method
------------------------------
    Function: onDispatch
       Class: Zend\Mvc\Controller\AbstractActionController
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-eventmanager/src/EventManager.php
        Line: 490
    Function: call_user_func
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-eventmanager/src/EventManager.php
        Line: 214
    Function: triggerListeners
       Class: Zend\EventManager\EventManager
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php
        Line: 118
    Function: trigger
       Class: Zend\EventManager\EventManager
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-mvc/src/DispatchListener.php
        Line: 93
    Function: dispatch
       Class: Zend\Mvc\Controller\AbstractController
        Type: method
------------------------------
    Function: onDispatch
       Class: Zend\Mvc\DispatchListener
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-eventmanager/src/EventManager.php
        Line: 490
    Function: call_user_func
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-eventmanager/src/EventManager.php
        Line: 214
    Function: triggerListeners
       Class: Zend\EventManager\EventManager
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/vendor/zendframework/zend-mvc/src/Application.php
        Line: 314
    Function: trigger
       Class: Zend\EventManager\EventManager
        Type: method
------------------------------
        File: /home/cbleek/Projects/YAWIK/public/index.php
        Line: 49
    Function: run
       Class: Zend\Mvc\Application
        Type: method
------------------------------

```

**Comment by cbleek** [comment]
The bug was introduced in 99270f8

**Comment by cbleek** [comment]
@fedys : before this commit, it was possible to login with google and github. Can you take a look at this?

**Comment by TiSiE** [comment]
I'm not sure, if we need the social profiles plugin for this feature. Using just the user profile provided by HybridAuth is sufficient. The former are really only needed if someone wants to attach the profile to an application (or CV) and will be fetched then again anyway.

**Comment by fedys** [comment]
@TiSiE you are right

**Comment by fedys** [comment]
@cbleek I am working on it.

**Comment by fedys** [comment]
There are two possible solutions:

1. To catch \InvalidArgumentException on the following [line](https://github.com/cross-solution/YAWIK/blob/develop/module/Auth/src/Auth/Adapter/HybridAuth.php#L110) and in such case leave the profile data empty. No other changes are needed.
2. To get rid off the SocialProfilePlugin dependency in the [HybridAuth](https://github.com/cross-solution/YAWIK/blob/develop/module/Auth/src/Auth/Adapter/HybridAuth.php) and change a logic of showing profile info on the page [/en/my/profile](https://github.com/cross-solution/YAWIK/blob/develop/module/Auth/src/Auth/Controller/ManageController.php#L49) . This involves changes in the social profiles form when an user clicks the "view profile data" link with profile data not  yet populated. In such case the SocialProfilePlugin has to be called and populate the data accordingly.

Which solution do you prefer?

**Comment by TiSiE** [comment]
Implement the first solution, to quickly fix the error for v0.26.

The feature was implemented reusing the SocialProfiles form, but as we now see, this was the wrong approach. Because attaching a profile to an application/cv and attaching a profile to a user are two completely different things. But we will address the needed rework in a new issue (for v0.27 or later).
